src/components/views/dialogs/UserSettingsDialog.tsx

```diff
@@ -43,32 +44,42 @@ interface IProps extends IDialogProps {
 
 interface IState {
     mjolnirEnabled: boolean;
+    newSessionManagerEnabled: boolean;
 }
 
 export default class UserSettingsDialog extends React.Component<IProps, IState> {
-    private mjolnirWatcher: string | undefined;
+    private settingsWatchers: string[] = [];
 
     constructor(props) {
         super(props);
 
         this.state = {
             mjolnirEnabled: SettingsStore.getValue("feature_mjolnir"),
+            newSessionManagerEnabled: SettingsStore.getValue("feature_new_device_manager"),
         };
     }
 
     public componentDidMount(): void {
-        this.mjolnirWatcher = SettingsStore.watchSetting("feature_mjolnir", null, this.mjolnirChanged);
+        this.settingsWatchers = [
+            SettingsStore.watchSetting("feature_mjolnir", null, this.mjolnirChanged),
+            SettingsStore.watchSetting("feature_new_device_manager", null, this.sessionManagerChanged),
+        ];
     }
 
     public componentWillUnmount(): void {
-        this.mjolnirWatcher && SettingsStore.unwatchSetting(this.mjolnirWatcher);
+        this.settingsWatchers.forEach(watcherRef => SettingsStore.unwatchSetting(watcherRef));
     }
 
     private mjolnirChanged: CallbackFn = (settingName, roomId, atLevel, newValue) => {
         // We can cheat because we know what levels a feature is tracked at, and how it is tracked
         this.setState({ mjolnirEnabled: newValue });
     };
 
+    private sessionManagerChanged: CallbackFn = (settingName, roomId, atLevel, newValue) => {
+        // We can cheat because we know what levels a feature is tracked at, and how it is tracked
+        this.setState({ newSessionManagerEnabled: newValue });
+    };
```

Comment:
The `TabView` uses a number to track which tab is being viewed, so if we push a new tab while viewing the settings it could cause a shift to a different tab. If that really happens, we should either force the user to re-open the dialog or fix the `TabView`

(at least this happened the last time I tried to do this, iirc)

Reply:
Good spot! Opened this PR to fix: https://github.com/matrix-org/matrix-react-sdk/pull/9145